pkg/metrics/providers/graphite.go

```diff
@@ -109,33 +170,45 @@ func (g *GraphiteProvider) RunQuery(query string) (graphiteResponse, error) {
 
 	r, err := http.DefaultClient.Do(req.WithContext(ctx))
 	if err != nil {
-		return nil, fmt.Errorf("request failed: %w", err)
+		return 0, fmt.Errorf("request failed: %w", err)
 	}
 	defer r.Body.Close()
 
 	b, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		return nil, fmt.Errorf("error reading body: %w", err)
+		return 0, fmt.Errorf("error reading body: %w", err)
 	}
 
 	if 400 <= r.StatusCode {
-		return nil, fmt.Errorf("error response: %s", string(b))
+		return 0, fmt.Errorf("error response: %s", string(b))
```

Comment:
This may not be too important, but I wonder if we should check `if 400 <= r.StatusCode` before attempting to `ioutil.ReadAll(r.Body)`.